connexion/api.py

```diff
@@ -84,12 +84,17 @@ def __init__(self, swagger_yaml_path, base_url=None, arguments=None,
                             'swagger_url': swagger_url,
                             'auth_all_paths': auth_all_paths})
         arguments = arguments or {}
-        with swagger_yaml_path.open() as swagger_yaml:
-            swagger_template = swagger_yaml.read()
+        with swagger_yaml_path.open(mode='rb') as swagger_yaml:
+            contents = swagger_yaml.read()
+            try:
+                swagger_template = contents.decode()
+            except UnicodeDecodeError:
+                swagger_template = contents.decode('utf-8', 'replace')
```

Comment:
Could you please provide an "encoding that's completely incompatible with ASCII/UTF-8" so we can add a test case for that and adapt the code to handle this situation?